### ISSUE TITLE
Unable to export to PDF

### ISSUE THREAD

**Issue by tintou** [CollaboraOnline/online#5325]
**Describe the bug**
Everytime that I try to export any document to PDF, the export fails, it works when I export to .doc or .rtf

**To Reproduce**
1. Go to 'Files'
2. Click on 'Download as' -> 'PDF Document'

I get a popup with "Document cannot be exported. please contact administrator"

In the logs, I'm getting:
```
kit-16392-16374 2022-10-08 21:22:02.643717 +0000 [ kitbroker_002 ] ERR  ToMaster-013: SaveAs Failed for id=export [/tmp/user/docs/BIqkZrHxI5Yxl7bC9nQvwqoWG6Yy4jBpRgvNWCtpy1aIg9yHAEHiHGysr3Enciqh/1%20-%20Atelier_proposition_CREIRA.pdf]. error= exception: SfxBaseModel::impl_store <file:///tmp/user/docs/BIqkZrHxI5Yxl7bC9nQvwqoWG6Yy4jBpRgvNWCtpy1aIg9yHAEHiHGysr3Enciqh/1%20-%20Atelier_proposition_CREIRA.pdf> failed: 0xc10(Error Area:Io Class:Write Code:16) /home/collabora/jenkins/workspace/core-co-22.05-for-arm64-PI/sfx2/source/doc/sfxbasemodel.cxx:3202 /home/collabora/jenkins/workspace/core-co-22.05-for-arm64-PI/sfx2/source/doc/sfxbasemodel.cxx:1775| kit/ChildSession.cpp:1090
```

Note that cool runs on an ARM64 machine.

**Expected behavior**
A PDF can be downloaded

**Actual behavior**
An Error

**Comment by Ezinnem** [comment]
@tintou , Please can you share the COOL version that you are using?

**Comment by vmiklos** [comment]
Is this with a self-built core.git? The Collabora-produced COOL binaries build core.git using --with-distro=CPLinux-LOKit, which includes --without-system-nss. With system nss, it's a known problem that PDF export can't work because it will fail to compute the checksum of the document on export.

**Comment by tintou** [comment]
my cool version: `22.05.6.3 git hash: abd9205e`

**Comment by tintou** [comment]
I'm using the arm64 packages from https://www.collaboraoffice.com/repos/CollaboraOnline/CODE-arm64-ubuntu1804/

**Comment by vmiklos** [comment]
OK, then this will be something else, those binaries already use --with-distro=CPLinux-LOKit.

**Comment by timar** [comment]
PDF export needs md5sum functions for creating hashes, and they are provided by nss. The error occurs when nss cannot be initialized. Usually it happens when /dev/random or /dev/urandom is not accessible (e.g. rootless containers). @tintou can it be your case?

**Comment by tintou** [comment]
I have md5sum installed and working, I'm not using a container

---

With the very latest update, the menu is greyed-out:

![Capture d'écran du 2022-12-29 11 56 38@2x](https://user-images.githubusercontent.com/745535/209941682-c8718f43-82e5-4dbb-9ceb-09c6e06dc92d.png)

Version COOLWSD :
22.05.9.2 git hash: [20c06f57](https://github.com/CollaboraOnline/online/commits/20c06f57)
Versison LOKit :
Collabora Office 22.05.9.2 git hash: [75263d0](https://hub.libreoffice.org/git-core/75263d0)

**Comment by gitgt** [comment]
Not clear why would this be different from https://github.com/CollaboraOnline/online/issues/5039? Can we close as a duplicate?

**Comment by sebastiaanveld** [comment]
I tested this today and found the issue is that exporting filenames with **spaces in the name** does not work. 
If I use a filename that has no spaces in the name exporting works fine. This is for both exporting to a PDF and EPUB format.
Document format used does not matter, using a .odt here, but also a docx has the same export issue if using spaces in the name.

Steps to duplicate:
- Create a new document in Nextcloud "Test export.odt"
- Open the document for editing
- Click "Export As"> PDF
- Click OK to accept export defaults
- Then the error appears :

![image](https://user-images.githubusercontent.com/37268999/224265905-7ccd7b76-688c-4207-97fb-ade2d9e252cd.png)

- Now click "Export As"> EPUB
- Click OK to accept export defaults
- Then the same error appears :

![image](https://user-images.githubusercontent.com/37268999/224266328-f1ca13b4-8b46-425b-8010-fcc722ae127d.png)

If you do the same for a filename without spaces in the name like "Test.odt" exporting works fine to either a PDF or EPUB format:

![image](https://user-images.githubusercontent.com/37268999/224266846-a59973d7-5750-4321-95a6-2ec4054e6a58.png)



Environment:
Nextcloud 25.0.4
COOLWSD version: 22.05.10.8 git hash:  [ef89c82]
LOKit version: Collabora Office 22.05.10.7 git hash: [d1b6e23]

**Comment by gitgt** [comment]
Report wasn't clear on spaces, as are not two other issues, but let's focus this one on spaces.

**Comment by gitgt** [comment]
There was a fix in COOLWSD version: 22.05.12.2. Please retest.

**Comment by sebastiaanveld** [comment]
Retested using build 22.05.12.2 and it works for me, thx!
